### ISSUE TITLE
[false-positive] SuspiciousBinaryOperationInspection

### ISSUE THREAD

**Issue by voku** [kalessil/phpinspectionsea#1246]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), v2.0.13 |
| Language level | PHP 7.0                        |

Current behaviour
--

error messge: ```'$price == 0' seems to be always false when reached.```

code:
```php
<?php

/**
 * @return null|string e.g. null, "0.00000", "3.42", ...
 */
function getPrice()
{
    // ...
    if (random_int(0, 1) === 1) {
        return null;
    }

    return "0.00000";
}

$price = getPrice();

if (
    $price == null
    ||
    $price == 0
) {
    // ...
    var_dump($price);
}
```

Expected behaviour
--
we need to keep "$price == null" and "$price == 0" ... related to this ticket: https://github.com/kalessil/phpinspectionsea/issues/1239#issuecomment-474079320

Environment details
---
PhpStorm 2018.3.5
Build #PS-183.5912.26, built on March 7, 2019
Licensed to Portable UTF-8 / Lars Moelleken
Subscription is active until February 28, 2020
For non-commercial open source development only.
JRE: 1.8.0_152-release-1343-b28 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.4.0-142-generic

**Comment by kalessil** [comment]
Checked, the behavior is expected. I can recommend 2 options here:
- suppression
- `$price === null || $price == 0`
